service/history/handler.go

```diff
@@ -197,11 +197,29 @@ func (h *Handler) isStopped() bool {
 	return atomic.LoadInt32(&h.status) == common.DaemonStatusStopped
 }
 
-// IsWorkflowTaskValid - whether workflow task is still valid
-func (h *Handler) IsWorkflowTaskValid(ctx context.Context, request *historyservice.IsWorkflowTaskValidRequest) (_ *historyservice.IsWorkflowTaskValidResponse, retError error) {
-	defer log.CapturePanic(h.logger, &retError)
+func lifecycleInterceptor(
+	ctx context.Context,
+	req interface{},
+	info *grpc.UnaryServerInfo,
+	handler grpc.UnaryHandler,
+) (_ interface{}, retErr error) {
+	h, _ := info.Server.(*Handler)
+
+	defer metrics.CapturePanic(h.logger, h.metricsHandler, &retErr)
```

Comment:
Let's make a separate interceptor for `CapturePanic`. It has nothing to do with "life cycle".